src/expr/impl/src/scalar/external/iceberg.rs

```diff
@@ -95,11 +98,10 @@ fn build(return_type: DataType, mut children: Vec<BoxedExpression>) -> Result<Bo
     // Check type:
     // 1. input type can be transform successfully
     // 2. return type is the same as the result type
-    let input_arrow_type = IcebergArrowConvert
-        .to_arrow_field("", &children[1].return_type())?
+    let input_arrow_type = to_arrow_field_for_iceberg("", &children[1].return_type())?
         .data_type()
         .clone();
-    let output_arrow_field = IcebergArrowConvert.to_arrow_field("", &return_type)?;
+    let output_arrow_field = to_arrow_field_for_iceberg("", &return_type)?;
```

Comment:
The style of `IcebergArrowConvert.to_arrow_field` looks better than `to_arrow_field_for_iceberg` and it is more consistent with the DeltaLake and UDF converters.

Reply:
Let's keep the style `IcebergArrowConvert.to_arrow_field`. But I avoid exposing the trait directly. I'm not sure whether is a good practice to expose it.